fix auto grad summing for IfOp where intermediate output needs renaming

harouwu opened the pull request:
Summary:
fix auto grad summing for IfOp where intermediate output needs renaming.

Bug before this diff:
- we only renames the output of IfOp without changing the subnet ops output
- this results in blob not found error

the unittest provides an example
this diff fix that for IfOp

Differential Revision: D13327090

Maratyszcza reviewed (approved):
dogscience.jpg

stamping to unblock

harouwu committed:
fix auto grad summing for IfOp where intermediate output needs renaming (#14772)

Summary:
fix auto grad summing for IfOp where intermediate output needs renaming.

Bug before this diff:
- we only renames the output of IfOp without changing the subnet ops output
- this results in blob not found error

the unittest provides an example
this diff fix that for IfOp
Pull Request resolved: https://github.com/pytorch/pytorch/pull/14772

Differential Revision: D13327090

fbshipit-source-id: 5332b2f04f9c4d2878a30a25cb750da1e0554998
